antsibull/write_docs.py

```diff
@@ -231,22 +240,27 @@ async def output_all_plugin_rst(collection_to_plugin_info: CollectionInfoT,
     env = doc_environment(('antsibull.data', 'docsite'))
     # Get the templates
     plugin_tmpl = env.get_template('plugin.rst.j2')
+    role_tmpl = env.get_template('role.rst.j2')
     error_tmpl = env.get_template('plugin-error.rst.j2')
 
     writers = []
     lib_ctx = app_context.lib_ctx.get()
     async with asyncio_pool.AioPool(size=lib_ctx.thread_max) as pool:
         for collection_name, plugins_by_type in collection_to_plugin_info.items():
             for plugin_type, plugins in plugins_by_type.items():
+                plugin_type_tmpl = plugin_tmpl
+                if plugin_type == 'role':
```

Comment:
This is probably the simplest way to extend the existing docs, but . .  . are roles actually plugins? If they are,  then this is fine. If they aren't, this seems like adding confusion for Future Coders. If roles aren't plugins, then could we use a different name? Like "elements" or something - where all plugins would be elements, and roles would also be elements . . . ?

Reply:
That's a good question, and nothing I can easily answer. I would see it similar to "are modules plugins?". This is also something @abadger wanted to look at.

Reply:
Yeah, the way that the ansible-core code works, modules really are a type of plugin.  roles are not.  I'm working up an alternate structure for felixfontein's code to see if it looks like it will be easier to maintain as a separate code path rather than pretending like they're a different type of plugin.  I'm not sure yet if it will be... have to see them side by side to decide.